ui/src/adapters/api/connections.ts

```diff
@@ -0,0 +1,85 @@
+import axios from "axios";
+import { z } from "zod";
+
+import { APIResponse, HTTPStatusSuccess, ResultOK, buildAPIError } from "src/adapters/api";
+import { Env } from "src/domain";
+import { buildAuthorizationHeader } from "src/utils/browser";
+import { API_VERSION, QUERY_SEARCH_PARAM } from "src/utils/constants";
+import { StrictSchema } from "src/utils/types";
+
+//TODO move it to credentials when is properly cleaned
+export interface Credential {
+  attributes: {
+    type: string;
+  };
+  id: string;
+}
+
+export interface Connection {
+  createdAt: string;
+  credentials: Credential[];
+  id: string;
+  issuerID: string;
+  userID: string;
+}
+
+export async function getConnections({
+  credentials,
+  env,
+  params: { query },
+  signal,
+}: {
+  credentials: boolean;
+  env: Env;
+  params: {
+    query?: string;
+  };
+  signal: AbortSignal;
+}): Promise<APIResponse<Connection[]>> {
+  try {
+    const response = await axios({
+      baseURL: env.api.url,
+      headers: {
+        Authorization: buildAuthorizationHeader(env),
+      },
+      method: "GET",
+      params: new URLSearchParams({
+        ...(query !== undefined ? { [QUERY_SEARCH_PARAM]: query } : {}),
+        ...(credentials ? { credentials: "true" } : {}),
+      }),
+      signal,
+      url: `${API_VERSION}/connections`,
+    });
+    const { data } = resultOKConnections.parse(response);
+
+    return { data, isSuccessful: true };
+  } catch (error) {
+    return { error: buildAPIError(error), isSuccessful: false };
```

Comment:
This pattern is not correct. We're losing track of the parsing errors. Will open an issue to fix it globally.